services/crawler/services/IotexService.ts

```diff
@@ -0,0 +1,206 @@
+import Antenna from 'iotex-antenna'
+import { IActionInfo, IGetBlockMetasResponse, IGetChainMetaResponse, IGetReceiptByActionResponse, IGetServerMetaResponse } from 'iotex-antenna/lib/rpc-method/types'
+import { from } from '@iotexproject/iotex-address-ts'
+import { CandidateRegister, CandidateUpdate, StakeRestake, StakeWithdraw, StakeUnstake, StakeAddDeposit, ClaimFromRewardingFund, StakeCreate, StakeTransferOwnership } from 'iotex-antenna/lib/action/types'
+import { DepositToRewardingFund } from 'iotex-antenna/protogen/proto/types/action_pb'
+
+export interface IotexServiceOptions {
+  network: IotexNetworkType
+}
+
+export enum IotexNetworkType {
+  Mainnet = 'mainnet',
+  Testnet = 'testnet'
+}
+
+export interface IotexBlock {
+  type: string
+  id: string
+  height: number
+  datestring: string
+  producer: string
+  amount: string
+  transaction_root: string
+  num_of_actions: number
+}
+
+export interface IotexGovernanaceActions {
+  grantReward: any[]
+  claimFromRewardingFund: ClaimFromRewardingFund[]
+  depositToRewardingFund: DepositToRewardingFund[]
+  candidateRegister: CandidateRegister[]
+  candidateUpdate: CandidateUpdate[]
+  createStake: StakeCreate[]
+  stakeRestake: StakeRestake[]
+  depositToStake: StakeAddDeposit[]
+  transferStake: StakeTransferOwnership[]
+  stakeUnstake: StakeUnstake[]
+  stakeWithdraw: StakeWithdraw[]
+}
+
+export interface CreateStakeTableColumns {
+  type: string
+  datestring: string
+  address: string
+  staked_candidate: string
+  staked_amount: number
+  staked_duration: number
+  auto_stake: boolean
+}
+
+export class IoTexService {
+  network: IotexNetworkType
+  endpoint: string
+  client: Antenna
+  constructor (opt: IotexServiceOptions) {
+    this.network = opt.network ?? IotexNetworkType.Mainnet
+    this.endpoint = this.network === IotexNetworkType.Mainnet ? 'https://api.iotex.one:443' : 'https://api.testnet.iotex.one:443'
+    this.client = new Antenna(this.endpoint)
```

Comment:
With the new iotex-antenna version (updated in the latest commit here), you'll need to pass the chain ID. 1 or 2 for mainnet or testnet, respectively.

Reply:
@hawyar